lib/cocoapods/generator/xcconfig/xcconfig_helper.rb

```diff
@@ -321,10 +321,20 @@ def self.settings_for_dependent_targets(target, dependent_targets, test_xcconfig
         #
         def self.generate_vendored_build_settings(aggregate_target, pod_targets, xcconfig)
           pod_targets.each do |pod_target|
-            unless pod_target.should_build? && pod_target.requires_frameworks?
+            unless pod_target.should_build? && pod_target.requires_frameworks? && !pod_target.static_framework?
               XCConfigHelper.add_settings_for_file_accessors_of_target(aggregate_target, pod_target, xcconfig)
             end
           end
+          # Make sure -framework option gets added for the search paths when static_frameworks are involved.
+          # Otherwise test targets won't link in their primary target's dependencies.
+          unless aggregate_target.nil?
+            dependent_targets = aggregate_target.search_paths_aggregate_targets
+            dependent_targets.each do |dependent_target|
```

Comment:
can do `dependent_target.select(&:static_framework?)`?

Reply:
No, because all the pod_targets need to be added whether or not they're a static_framework. 

I don't think the static_framework? check is needed at all, but I want to avoid changing projects that aren't using it to be conservative.